spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/GatewayApiController.java

```diff
@@ -51,4 +69,34 @@ public ApiKey findApiKeyEntityByApiKeyValue(@RequestParam("keyValue") String api
     public ServiceApi findServiceByApiKeyValue(@RequestParam("apiKeyValue") String apiKeyValue) {
         return serviceApiRepository.findByApiKeyValue(apiKeyValue);
     }
+
+    @GetMapping( "/api/gateways/api-keys")
+    public ResponseEntity<List<ApiKey>> findApiKeyListByGatewayApiKey(@RequestParam("gateway-api-key") String gatewayApiKey) {
+        log.debug("The gateway with the ID {} requests its latest API key list.", gatewayApiKey);
+        try {
+            Proxy proxy = proxyRepository.findById(gatewayApiKey);
+            if (proxy != null) {
+                proxy.setLastCallTimeToAdmin(new Timestamp(System.currentTimeMillis()).getTime());
```

Comment:
In general, you should use Java's newer Date/Time API: https://www.baeldung.com/java-8-date-time-intro which has several advantages over Date and Timestamp. This API is also recommended by @gr-hovest-atb.

Reply:
I agree and done